modules/product/src/Plugin/Field/FieldWidget/ProductVariationTitleWidget.php

```diff
@@ -153,6 +156,9 @@ public function formElement(FieldItemListInterface $items, $delta, array $elemen
    */
   protected function selectVariationFromUserInput(array $variations, array $user_input) {
     $current_variation = NULL;
+    if (!isset($variations[$user_input['variation']])) {
```

Comment:
The `!empty($user_input['variation'])` check below should cover this `isset`. `empty` handles not set as well.

Reply:
May be instead of all the above checks for empty user input (and therefore empty selected variation) we just replicate logic from here:
https://github.com/drupalcommerce/commerce/blob/8.x-2.x/modules/product/src/Plugin/Field/FieldWidget/ProductVariationAttributesWidget.php#L124

```
if ($form_state->isRebuilding()) {
  $parents = array_merge($element['#field_parents'], [$items->getName(), $delta]);
  $user_input = (array) NestedArray::getValue($form_state->getUserInput(), $parents);
  $selected_variation = $this->selectVariationFromUserInput($variations, $user_input);
}
```

So, if the form is rebuilding, which means it is not initial page loading, then take user input which cannot be empty at this moment as the rebuilding itself is initiated by a user input. For me it works flawlessly.